src/FileWriterTask.h

```diff
@@ -25,48 +23,90 @@ class ParseError : public std::runtime_error {
 /// and makes those available to the FileMaster and Streamer.
 /// Created by Master on command message and passed to FileMaster in ctor.
 class FileWriterTask final {
-  friend class ::Test___FileWriterTask___Create01;
-  friend class CommandHandler;
-
 public:
-  FileWriterTask(std::string ServiceID,
+  /// Constructor
+  ///
+  /// \param ServiceID The service ID.
+  /// \param StatusProducer_ The status producer.
+  FileWriterTask(std::string const &ServiceID,
                  std::shared_ptr<KafkaW::ProducerTopic> StatusProducer_);
+
+  /// Destructor.
   ~FileWriterTask();
-  FileWriterTask &set_hdf_filename(std::string hdf_output_prefix,
-                                   std::string hdf_filename);
-  /// Used by Streamer to get the list of demuxers
+
+  /// Initialise the HDF file.
+  ///
+  /// \param NexusStructure The structure of the NeXus file.
+  /// \param ConfigFile The configuration information.
+  /// \param HdfInfo The HDF information for the stream.
+  /// \param UseSwmr Whether to use SWMR.
+  void InitialiseHdf(std::string const &NexusStructure,
+                     std::string const &ConfigFile,
+                     std::vector<StreamHDFInfo> &HdfInfo, bool UseSwmr);
+
+  /// \brief  Set the `JobID`.
+  ///
+  /// \param Id The Id value to use.
+  void setJobId(const std::string &Id);
+
+  /// \brief Add a source to the demuxers.
+  ///
+  /// \param source The source to add.
+  void addSource(Source &&Source);
```

Comment:
Naming things is hard and one of the downsides of the LLVM standard is that it makes it even harder. Still, I think we can do better. I would have preferred to rename the type (class) `Source` to something else but that is not really an option for this pull request.
```suggestion
  void addSource(Source &&DataSource);
```